shared/utility/math/quaternion.hpp

```diff
@@ -48,16 +51,73 @@ namespace utility::math::quaternion {
         }
 
         // Scale the accumulator matrix
-        A /= std::distance(begin, end);
+        A /= static_cast<Scalar>(std::distance(begin, end));
 
         // Solve for the eigenvectors of the accumulator matrix
         Eigen::SelfAdjointEigenSolver<Eigen::Matrix<Scalar, 4, 4>> eigensolver(A);
         if (eigensolver.info() != Eigen::Success) {
-            throw std::runtime_error("Eigen decomposition failed");
+            // Failed to get the mean using Merkley's method, try just averaging naively
+            Eigen::Vector4d mean = Eigen::Vector4d::Zero();
+            for (Iterator it = begin; it != end; ++it) {
+                mean += it->coeffs();
```

Comment:
This one could definitely be an accumulate, but I won't request changes on that behalf :p